backend/recommendation.js

```diff
@@ -62,24 +67,22 @@ async function recommendEventSlots(userId, duration, targetDate, invitees = [])
 
         let timeGrid = generateTimeGrid();
         users.forEach(user => {
-            const userTimeGrid = generateTimeGrid();
             user.calendarEvents.forEach(event => {
                 if (event.status === 'accepted') {
-                    markUnavailableSlots(userTimeGrid, new Date(event.startAt), new Date(event.endAt));
+                    markUnavailableSlots(timeGrid, new Date(event.startAt), new Date(event.endAt));
                 }
             });
             for (let i = 0; i < user.preferredStartTime; i++) {
-                userTimeGrid[i].available = false;
+                timeGrid[i].available = false;
             }
-            for (let i = user.preferredEndTime; i < userTimeGrid.length; i++) {
-                userTimeGrid[i].available = false;
+            for (let i = user.preferredEndTime; i < timeGrid.length; i++) {
+                timeGrid[i].available = false;
             }
-            timeGrid = timeGrid.map((slot, index) => ({ slot: index, available: slot.available && userTimeGrid[index].available }));
         });
 
         const availableSlots = findAvailableSlots(timeGrid, duration);
         if (availableSlots.length === 0) {
-            throw new Error('No available slots found');
+            return -1;
         }
```

Comment:
Introducing this -1 value just adds confusion, it would be clearer to just handle the case that the length is 0 in the frontend.